### PR TITLE
[search] Implemented batching of search results.

### PR DESCRIPTION
**Pull request by mpimenov**
@ygorshenin PTAL

### CODE REVIEW

File: search/ranker.cpp [hunk 1]
```diff
@@ -477,56 +483,64 @@ void Ranker::ProcessSuggestions(vector<IndexedValue> & vec, Results & res) const
...
-  MakePreResult2(geocoderParams, values, streets);
-  RemoveDuplicatingLinear(values);
-  if (values.empty())
-    return;
-
-  sort(values.begin(), values.end(), my::LessBy(&IndexedValue::GetDistanceToPivot));
+    if (m_params.m_viewportSearch)
+    {
+      m_results.AddResultNoChecks(
+          (*(values[i]))
```
Comment (mgsergio) [inline search/ranker.cpp]: *values[i]

Comment (mpimenov) [inline search/ranker.cpp]: There's no overloaded ```->```. I removed a pair of redundant brackets, however.

**Comment by mpimenov** [comment]
PTAL

**Comment by mgsergio** [comment]
LGTM

File: search/pre_ranker.hpp [hunk 2]
```diff
@@ -34,9 +34,16 @@ class PreRanker
     int m_scale = 0;
   };
 
+  static size_t const kBatchSize;
```
Comment (ygorshenin) [inline search/pre_ranker.hpp]: Do we need this in public section?

File: search/pre_ranker.hpp [hunk 3]
```diff
@@ -34,9 +34,16 @@ class PreRanker
...
+
   PreRanker(Index const & index, Ranker & ranker, size_t limit);
 
-  inline void Init(Params const & params) { m_params = params; }
+  inline void Init(Params const & params)
```
Comment (ygorshenin) [inline search/pre_ranker.hpp]: Please, don't inline methods that are in two or more lines.

File: search/ranker.cpp [hunk 4]
```diff
@@ -130,6 +131,9 @@ class CombinedIter
 };
 }  // namespace
 
+// static
+size_t const Ranker::kBatchSize = 10;
```
Comment (ygorshenin) [inline search/ranker.cpp]: Please, move this to the Ranker definitions block.

File: search/ranker.cpp [hunk 5]
```diff
@@ -477,56 +483,64 @@ void Ranker::ProcessSuggestions(vector<IndexedValue> & vec, Results & res) const
...
+        ++count;
+    }
   }
+  move(values.begin() + i, values.end(), back_inserter(m_leftovers));
+
+  m_preResults1.clear();
+  m_params.m_onResults(m_results);
 }
 
+void Ranker::FlushResults() { UpdateResults(true /* lastUpdate */); }
```
Comment (ygorshenin) [inline search/ranker.cpp]: This method may be inlined. Or leave it as is, but add a blank line after it.

**Comment by ygorshenin** [comment]
LGTM, but I don't like the idea to move IndexedValues from m_leftovers to local, and at the end from local back to m_leftovers. Consider to fix it.

File: search/ranker.cpp (hunk 5 above)
Comment (mpimenov) [inline search/ranker.cpp]: done

File: search/ranker.cpp (hunk 4 above)
Comment (mpimenov) [inline search/ranker.cpp]: done

File: search/pre_ranker.hpp (hunk 3 above)
Comment (mpimenov) [inline search/pre_ranker.hpp]: done

File: search/pre_ranker.hpp (hunk 2 above)
Comment (mpimenov) [inline search/pre_ranker.hpp]: Not really.
done

**Comment by mpimenov** [comment]
I changed the m_leftovers part as per your suggestion.
PTAL

**Comment by ygorshenin** [comment]
LGTM

**Pull request merged**